test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -210,10 +213,11 @@ def world_size(self):
     @common_distributed.skip_if_rocm
     def test_step(self):
         """ Check that the ZeroRedundancyOptimizer wrapper properly exposes the `.step()` interface"""
-        if self.rank > 1 or (BACKEND == dist.Backend.NCCL and torch.cuda.device_count() < 2):
+
+        if self.rank >= self.world_size or (BACKEND == dist.Backend.NCCL and torch.cuda.device_count() < 2):
```

Comment:
I suspect that previously having the function `return` if `self.rank > 1` was meant to avoid cases where `world_size` is greater than `2`. However, it led to the test timing out (i.e. hanging) since those with `self.rank` equal to `0` and `1` would still proceed and be waiting at a barrier in `dist.init_process_group()`.